chain/src/chain.rs

```diff
@@ -791,11 +791,11 @@ impl Chain {
 		let mut oldest_height = 0;
 		let mut oldest_hash = ZERO_HASH;
 
-		let mut current = self.get_block_header(&header_head.last_block_h);
+		let mut current = self.get_block_header(&body_head.last_block_h);
```

Comment:
Lets rename _this_ usage of current to make it clearer that we want to check for headers in our db based on `body_head`.

This is a 2 step process.

* start with `head` (head of the full block chain)
  * if we do not have a _full block_ in the db for this `head` then its an error
  * if we do not have the header in the db for this `head` then its an error
* traverse back through the full block chain from `head` until we find a header that "is on current chain"
  * this is our "fork point" between existing header chain and full block chain
* now we want to traverse back through our header chain from `header_head` back to this fork point
  * these are the blocks that we need to request (we have the header but not the full block)

Reply:
I put some of yours comments to the source code comments, and renamed the 2nd `current`. 
pls let me whether it's better now.